knowledge_graph/scoring.py

```diff
@@ -0,0 +1,57 @@
+# Let's start by creating a function that just extracts candidate DNEs
+# and define our skeleton ranking function as just throwing runifs at
+# candidates at returning the top_k requested, with the associated score 
+# and rank
+
+import networkx as nx
+import random
+
+# Should probably import this from somewhere general purpose
+def load_ontology(fpath='knowledge_graph/Climate_Mind_DiGraph.gpickle'):
+    return nx.read_gpickle(fpath)
+
+class NodeRecommender:
+    def __init__(self, 
+                 g: nx.DiGraph):
+        self.g = g
+    @property
+    def candidates(self):
+        if not hasattribute(self, '_candidates'):
+            self._candidates = self.get_scoring_candidates(self.g)
+        return self._candidates
+    @staticmethod
+    def get_node_scoring_candidates(g):
+        candidates = []
+        for node_id, node_data in g.nodes(data=True):
+            if 'downstream negative effect' in node_data['classes']:
```

Comment:
Small thing, but should these be extracted as constants instead of being free text in the code?